File: app/src/main/java/com/montreal/wtm/model/Sponsor.java
package com.montreal.wtm.model;

import com.google.firebase.database.IgnoreExtraProperties;
import com.google.firebase.database.PropertyName;

@IgnoreExtraProperties
public class Sponsor {

    @PropertyName("image")
    public String imageKey;

    @PropertyName("name")
    public String name;

    @PropertyName("website")
    public String urlWebsite;

    public Sponsor() {
    }

    public Sponsor(String imageKey, String name, String urlWebsite) {
        this.imageKey = imageKey;
        this.name = name;
        this.urlWebsite = urlWebsite;
    }

    public String getImageKey() {
        return imageKey;
    }

    public String getName() {
        return name;
    }

    public String getUrlWebsite() {
        return urlWebsite;
    }
}

File: app/src/main/java/com/montreal/wtm/model/Sponsor.kt
package com.montreal.wtm.model

import android.os.Parcel
import android.os.Parcelable
import com.google.firebase.database.IgnoreExtraProperties
import com.google.firebase.database.PropertyName

@IgnoreExtraProperties
class Sponsor : Parcelable {

  @PropertyName("logoUrl")
  var logoUrl: String

  @PropertyName("name")
  var name: String

  @PropertyName("url")
  var urlWebsite: String

  override fun describeContents(): Int {
    return 0
  }

  override fun writeToParcel(dest: Parcel, flags: Int) {
    dest.writeString(this.logoUrl)
    dest.writeString(this.name)
    dest.writeString(this.urlWebsite)
  }

  protected constructor(`in`: Parcel) {
    this.logoUrl = `in`.readString()
    this.name = `in`.readString()
    this.urlWebsite = `in`.readString()
  }

  companion object {

    val CREATOR: Parcelable.Creator<Sponsor> = object : Parcelable.Creator<Sponsor> {
      override fun createFromParcel(source: Parcel): Sponsor {
        return Sponsor(source)
      }

      override fun newArray(size: Int): Array<Sponsor?> {
        return arrayOfNulls(size)
      }
    }
  }
}
